metricflow/specs/where_filter_transform.py

```diff
@@ -67,23 +67,41 @@ def create_from_where_filter(self, where_filter: WhereFilter) -> WhereFilterSpec
                 f"Error while rendering Jinja template:\n{where_filter.where_sql_template}"
             ) from e
 
-        """
-        Dimensions that are created with a grain parameter, Dimension(...).grain(...), are
-        added to dimension_specs otherwise they are add to time_dimension_factory.time_dimension_specs
-        """
-        dimension_specs = []
-        for dimension in dimension_factory.created:
-            if dimension.time_dimension_spec:
-                time_dimension_factory.time_dimension_specs.append(dimension.time_dimension_spec)
-            else:
-                dimension_specs.append(dimension.dimension_spec)
-
-        return WhereFilterSpec(
-            where_sql=rendered_sql_template,
-            bind_parameters=self._bind_parameters,
-            linkable_spec_set=LinkableSpecSet(
-                dimension_specs=tuple(dimension_specs),
-                time_dimension_specs=tuple(time_dimension_factory.time_dimension_specs),
-                entity_specs=tuple(entity_factory.entity_specs),
-            ),
-        )
+    def create_from_where_filter(self, where_filter: WhereFilter) -> WhereFilterSpec:  # noqa: D
+        # Used to check that call parameter sets generated by DSI match those generated below.
```

Comment:
nit: this comment is hard to understand.

Can we replace it, and the triple-quoted inline below, with a docstring for this method that explains why it's doing what it's doing? It's super odd to me to see a `#noqa: D` on a method annotated with a bunch of inline comments, including one in triple-quoted blocks, when most of those are describing the purpose of the function.

I realize none of this is due to your change, but while we are refactoring a bit updating and improving documentation on this method seems in bounds to me.